java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -0,0 +1,330 @@
+				return true;
+			}
+		});
+
+		weightTile = (LinearLayout) findViewById(R.id.WeightTile);
+		weightTile.setOnClickListener(new View.OnClickListener() {
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (service.isAppConnected()) {
+					intent = new Intent(MainActivity.this, WeightActivity.class);
+				}
+				else
+				{
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
+					startActivity(intent);
+				}
+			}
+		});
+	}
+	
+	@Override
+	protected void onStart() {
+		super.onStart();
+		hvClient.start();
+	}
+	
+	@Override
+    protected void onResume()
+    {
+		super.onResume();
+    }
+	
+    @Override
+	protected void onStop() {
+		super.onStop();
+		if(connectProgressDialog != null) {
+			connectProgressDialog.dismiss();
+		}
+		
+		hvClient.stop();
+	}
+    
+    @Override
+	public boolean onCreateOptionsMenu(Menu menu) {
+		getMenuInflater().inflate(R.menu.main, menu);
+		RelativeLayout recordNameLayout = (RelativeLayout) menu.findItem(R.id.record_name_layout).getActionView();
+        TextView tv = (TextView) recordNameLayout.findViewById(R.id.current_record_name);
+        
+        HealthVaultApp application = HealthVaultApp.getInstance();
+		if(service.isAppConnected() && application.getCurrentRecord() != null) {	
+	       tv.setText(application.getCurrentRecord().getName());
+	       
+	       recordNameLayout.setOnClickListener(new OnClickListener() {
+				
+				@Override
+				public void onClick(View arg0) {
+					Intent intent = new Intent(MainActivity.this, RecordPickerActivity.class);
+					startActivity(intent);
+				}
+			});
+		}
+		else {
+			tv.setVisibility(View.GONE);
+		}
+		
+		return true;
+	}
+
+	public void onConnected() {
+	}
+
+	public void onError(Exception e) {
+		Toast.makeText(
+                MainActivity.this, 
+                e.getMessage(), 
+                Toast.LENGTH_LONG).show();
+	}
+
+	@Override
+	public boolean onOptionsItemSelected(MenuItem item) {
+	    switch (item.getItemId()) {
+	        case R.id.action_connect:
+	            doConnect();
+	            return true;
+	        default:
+	            return super.onOptionsItemSelected(item);
+	    }
+	}
+	
+	private void doConnect() {
+		if (!service.isAppConnected()) {
+			connectProgressDialog = ProgressDialog.show(
+                    MainActivity.this,
+                    "",
+                    "Please wait...", 
+                    true);
+			
+			HealthVaultSettings settings = service.getSettings();
+			settings.setMasterAppId("c6ba979f-c342-4408-a2bc-0dfb43b2bf8d");
+			settings.setServiceUrl("https://platform.healthvault-ppe.com/platform/wildcat.ashx");
+			settings.setShellUrl("https://account.healthvault-ppe.com");
+			settings.setIsMultiInstanceAware(true);
+			settings.setIsMRA(true);
+			service.start(MainActivity.this, MainActivity.this);
+    	}
+		else {
+			Toast.makeText(
```

Comment:
I am unsure of what the line length limits are, but this probably doesn't need to be separated